test/runnable/xtest46.d

```diff
@@ -4175,10 +4175,10 @@ void test1962()
 /***************************************************/
 // https://issues.dlang.org/show_bug.cgi?id=6228
 
-
 void test6228()
 {
-    const(int)* ptr;
+    int val;
+    const(int)* ptr = &val;
```

Comment:
This seems odd, is this test not called? Because the original seems like it should segfault.

Reply:
This test is called.  The reason it doesn't segfault is because the operation is rewritten in such a way that the dereference never happens.

```
// Original.
const(int)* ptr = null;
const(int) temp = 0;
auto x = (*ptr) ^^ temp;

// Constant propagation
const(int)* ptr = null;
auto x = (*ptr) ^^ 0;

// Optimize PowExp x ^^ 0 -> (x, 1)
const(int)* ptr = null;
auto x = (*ptr, 1);

// Optimize CommaExp removes expressions free of side effects.
auto x = 1 
```

Reply:
To further clarify, dmd is does not appear to optimize/dce the call to std.math.pow.  Unlike for example gdc...